pyasdf/asdftypes.py

```diff
@@ -108,13 +109,37 @@ def from_yaml_tag(self, tag):
 
     @lru_cache(5)
     def has_hook(self, hook_name):
+        """
+        Returns `True` if the given hook name exists on any of the managed
+        types.
+        """
         for cls in self._all_types:
             if hasattr(cls, hook_name):
                 return True
         return False
```

Comment:
I see now.  That sounds really useful.  So the reason the `lru_cache` only has 5 entries anyways is because there are only so many hook types?

Is there a list somewhere of all the hook names? Or is it just whatever names get passed at various to points to `AsdfFile.run_hook`?